src/operator/contrib/two_bit_quantize.cc

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file two_bit_quantize.cc
+ * \brief registers quantize_2bit, dequantize_2bit
+ * and create_2bit operators with nnvm
+ */
+#include "./two_bit_quantize-inl.h"
+
+namespace mxnet {
+namespace op {
+
+DMLC_REGISTER_PARAMETER(TwoBitParam);
+
+NNVM_REGISTER_OP(_contrib_quantize_2bit)
+.describe(R"code(Quantize an input tensor into using 2bits for each value using
+user-specified thresholds, while storing quantization error in residual array.
+
+The quantize_2bit operator takes 5 arguments and is called as follows:
+`quantize_2bit(array, residual, out, neg_threshold, pos_threshold)`.
```

Comment:
why not out = quantize_2bit(weight, residual)?

Reply:
Also I don't see why these need to be exposed as operators. It looks like for now kvstore handles everything internally.

Reply:
Do you think it can make it more flexible if the user wants to use the operator? It would only make sense if the user does gradient communication themselves, since these operators are specific because of the use of residual. I imagine that's not done by anyone, should I not register them as operators?

Reply:
If you do quantize manually then you can't use kvstore because it doesn't know the encoding of the gradients.
I think we should remove these operators for now and possibly add back when there is a better defined use case